modules/gnome-extensions/README.md

```diff
@@ -0,0 +1,24 @@
+# `gnome-extensions`
+
+The `gnome-extensions` module can be used to install Gnome extensions inside system directory.
+
+This module is universally compatible with all distributions which ship Gnome, as long as some extension doesn't require some additional dependency, like Pano.
+
+Thanks to https://extensions.gnome.org which provides end-releases of extensions as zips, it is very easy to maintain this module configuration.
+Basically the only maintenance is to bump the extension version when new Fedora/Gnome releases (around every 6 months).
```

Comment:
This sparked me to do some inspecting and debugger digging on the extensions page. I was able to find that an extension page on https://extensions.gnome.org/ contains a div with the data of all the versions, for example: 
```html
<div class="extension single-page dummy" data-epk="6" data-uuid="apps-menu@gnome-shell-extensions.gcampax.github.com" data-svm="{'3.2': {'pk': 284, 'version': 5}, '3.2.0': {'pk': 6, 'version': 1}, '3.3.2': {'pk': 33, 'version': 3}, '3.2.1': {'pk': 52, 'version': 4}, '3.2.3': {'pk': 284, 'version': 5}, '3.3.4': {'pk': 640, 'version': 7}, '3.3.90': {'pk': 831, 'version': 8}, '3.4': {'pk': 1008, 'version': 9}, '3.5.5': {'pk': 1690, 'version': 10}, '3.5.90': {'pk': 1736, 'version': 11}, '3.5.91': {'pk': 1807, 'version': 12}, '3.6': {'pk': 2213, 'version': 14}, '3.7.5': {'pk': 2489, 'version': 16}, '3.7.90': {'pk': 2560, 'version': 17}, '3.7.92': {'pk': 2640, 'version': 18}, '3.8': {'pk': 3216, 'version': 21}, '3.10': {'pk': 3247, 'version': 22}, '3.12': {'pk': 3739, 'version': 23}, '3.14': {'pk': 4332, 'version': 24}, '3.15.1': {'pk': 4463, 'version': 25}, '3.15.3': {'pk': 4582, 'version': 26}, '3.16': {'pk': 4763, 'version': 27}, '3.18': {'pk': 5320, 'version': 29}, '3.20': {'pk': 5764, 'version': 30}, '3.21.2': {'pk': 5980, 'version': 31}, '3.22': {'pk': 6209, 'version': 32}, '3.24': {'pk': 6867, 'version': 34}, '3.26': {'pk': 7470, 'version': 36}, '3.28': {'pk': 8121, 'version': 38}, '3.30': {'pk': 8381, 'version': 39}, '3.32': {'pk': 10201, 'version': 41}, '3.34': {'pk': 13363, 'version': 43}, '3.36': {'pk': 14372, 'version': 44}, '3.38': {'pk': 21648, 'version': 46}, '40.0': {'pk': 22834, 'version': 47}, '40': {'pk': 24975, 'version': 48}, '41': {'pk': 28466, 'version': 50}, '42': {'pk': 29906, 'version': 51}, '43': {'pk': 34582, 'version': 52}, '44': {'pk': 40512, 'version': 53}, '45': {'pk': 48949, 'version': 56}, '46': {'pk': 52131, 'version': 58}}" data-versions="{'3.2': {'6': {'pk': 6, 'version': '1'}, '52': {'pk': 52, 'version': '4'}, '284': {'pk': 284, 'version': '5'}}, '3.2.0': {'6': {'pk': 6, 'version': '1'}}, '3.3.2': {'33': {'pk': 33, 'version': '3'}}, '3.2.1': {'52': {'pk': 52, 'version': '4'}}, '3.2.3': {'284': {'pk': 284, 'version': '5'}}, '3.3.4': {'640': {'pk': 640, 'version': '7'}}, '3.3.90': {'831': {'pk': 831, 'version': '8'}}, '3.4': {'1008': {'pk': 1008, 'version': '9'}}, '3.5.5': {'1690': {'pk': 1690, 'version': '10'}}, '3.5.90': {'1736': {'pk': 1736, 'version': '11'}}, '3.5.91': {'1807': {'pk': 1807, 'version': '12'}}, '3.6': {'1972': {'pk': 1972, 'version': '13'}, '2213': {'pk': 2213, 'version': '14'}}, '3.7.5': {'2473': {'pk': 2473, 'version': '15'}, '2489': {'pk': 2489, 'version': '16'}}, '3.7.90': {'2560': {'pk': 2560, 'version': '17'}}, '3.7.92': {'2640': {'pk': 2640, 'version': '18'}}, '3.8': {'2691': {'pk': 2691, 'version': '19'}, '2971': {'pk': 2971, 'version': '20'}, '3216': {'pk': 3216, 'version': '21'}}, '3.10': {'3247': {'pk': 3247, 'version': '22'}}, '3.12': {'3739': {'pk': 3739, 'version': '23'}}, '3.14': {'4332': {'pk': 4332, 'version': '24'}}, '3.15.1': {'4463': {'pk': 4463, 'version': '25'}}, '3.15.3': {'4582': {'pk': 4582, 'version': '26'}}, '3.16': {'4763': {'pk': 4763, 'version': '27'}}, '3.18': {'5205': {'pk': 5205, 'version': '28'}, '5320': {'pk': 5320, 'version': '29'}}, '3.20': {'5764': {'pk': 5764, 'version': '30'}}, '3.21.2': {'5980': {'pk': 5980, 'version': '31'}}, '3.22': {'6209': {'pk': 6209, 'version': '32'}}, '3.24': {'6684': {'pk': 6684, 'version': '33'}, '6867': {'pk': 6867, 'version': '34'}}, '3.26': {'7416': {'pk': 7416, 'version': '35'}, '7470': {'pk': 7470, 'version': '36'}}, '3.28': {'8110': {'pk': 8110, 'version': '37'}, '8121': {'pk': 8121, 'version': '38'}}, '3.30': {'8381': {'pk': 8381, 'version': '39'}}, '3.32': {'9396': {'pk': 9396, 'version': '40'}, '10201': {'pk': 10201, 'version': '41'}}, '3.34': {'11903': {'pk': 11903, 'version': '42'}, '13363': {'pk': 13363, 'version': '43'}}, '3.36': {'14372': {'pk': 14372, 'version': '44'}}, '3.38': {'19636': {'pk': 19636, 'version': '45'}, '21648': {'pk': 21648, 'version': '46'}}, '40.0': {'22834': {'pk': 22834, 'version': '47'}}, '40': {'24975': {'pk': 24975, 'version': '48'}}, '41': {'26942': {'pk': 26942, 'version': '49'}, '28466': {'pk': 28466, 'version': '50'}}, '42': {'29906': {'pk': 29906, 'version': '51'}}, '43': {'34582': {'pk': 34582, 'version': '52'}}, '44': {'40512': {'pk': 40512, 'version': '53'}}, '45': {'43704': {'pk': 43704, 'version': '54'}, '44582': {'pk': 44582, 'version': '55'}, '48949': {'pk': 48949, 'version': '56'}}, '46': {'51672': {'pk': 51672, 'version': '57'}, '52131': {'pk': 52131, 'version': '58'}}}">
```

Don't know if this is useful, though, parsing HTML in bash is a lost cause.

Reply:
Actually, I was able to get that `data-svm` attribute with `xmllint` & `xpath` from a locally downloaded `.html` file.

`xmllint --html --xpath "/html/body//div[contains(@class, 'extension')]/@data-svm" ext6.html`

The output is still somewhat tough to parse, though.
```
ext6.html:40: HTML parser error : Tag header invalid
    <header class="gnome-header">
                                ^
 data-svm="{&quot;3.2&quot;: {&quot;pk&quot;: 284, &quot;version&quot;: 5}, &quot;3.2.0&quot;: {&quot;pk&quot;: 6, &quot;version&quot;: 1}, &quot;3.3.2&quot;: {&quot;pk&quot;: 33, &quot;version&quot;: 3}, &quot;3.2.1&quot;: {&quot;pk&quot;: 52, &quot;version&quot;: 4}, &quot;3.2.3&quot;: {&quot;pk&quot;: 284, &quot;version&quot;: 5}, &quot;3.3.4&quot;: {&quot;pk&quot;: 640, &quot;version&quot;: 7}, &quot;3.3.90&quot;: {&quot;pk&quot;: 831, &quot;version&quot;: 8}, &quot;3.4&quot;: {&quot;pk&quot;: 1008, &quot;version&quot;: 9}, &quot;3.5.5&quot;: {&quot;pk&quot;: 1690, &quot;version&quot;: 10}, &quot;3.5.90&quot;: {&quot;pk&quot;: 1736, &quot;version&quot;: 11}, &quot;3.5.91&quot;: {&quot;pk&quot;: 1807, &quot;version&quot;: 12}, &quot;3.6&quot;: {&quot;pk&quot;: 2213, &quot;version&quot;: 14}, &quot;3.7.5&quot;: {&quot;pk&quot;: 2489, &quot;version&quot;: 16}, &quot;3.7.90&quot;: {&quot;pk&quot;: 2560, &quot;version&quot;: 17}, &quot;3.7.92&quot;: {&quot;pk&quot;: 2640, &quot;version&quot;: 18}, &quot;3.8&quot;: {&quot;pk&quot;: 3216, &quot;version&quot;: 21}, &quot;3.10&quot;: {&quot;pk&quot;: 3247, &quot;version&quot;: 22}, &quot;3.12&quot;: {&quot;pk&quot;: 3739, &quot;version&quot;: 23}, &quot;3.14&quot;: {&quot;pk&quot;: 4332, &quot;version&quot;: 24}, &quot;3.15.1&quot;: {&quot;pk&quot;: 4463, &quot;version&quot;: 25}, &quot;3.15.3&quot;: {&quot;pk&quot;: 4582, &quot;version&quot;: 26}, &quot;3.16&quot;: {&quot;pk&quot;: 4763, &quot;version&quot;: 27}, &quot;3.18&quot;: {&quot;pk&quot;: 5320, &quot;version&quot;: 29}, &quot;3.20&quot;: {&quot;pk&quot;: 5764, &quot;version&quot;: 30}, &quot;3.21.2&quot;: {&quot;pk&quot;: 5980, &quot;version&quot;: 31}, &quot;3.22&quot;: {&quot;pk&quot;: 6209, &quot;version&quot;: 32}, &quot;3.24&quot;: {&quot;pk&quot;: 6867, &quot;version&quot;: 34}, &quot;3.26&quot;: {&quot;pk&quot;: 7470, &quot;version&quot;: 36}, &quot;3.28&quot;: {&quot;pk&quot;: 8121, &quot;version&quot;: 38}, &quot;3.30&quot;: {&quot;pk&quot;: 8381, &quot;version&quot;: 39}, &quot;3.32&quot;: {&quot;pk&quot;: 10201, &quot;version&quot;: 41}, &quot;3.34&quot;: {&quot;pk&quot;: 13363, &quot;version&quot;: 43}, &quot;3.36&quot;: {&quot;pk&quot;: 14372, &quot;version&quot;: 44}, &quot;3.38&quot;: {&quot;pk&quot;: 21648, &quot;version&quot;: 46}, &quot;40.0&quot;: {&quot;pk&quot;: 22834, &quot;version&quot;: 47}, &quot;40&quot;: {&quot;pk&quot;: 24975, &quot;version&quot;: 48}, &quot;41&quot;: {&quot;pk&quot;: 28466, &quot;version&quot;: 50}, &quot;42&quot;: {&quot;pk&quot;: 29906, &quot;version&quot;: 51}, &quot;43&quot;: {&quot;pk&quot;: 34582, &quot;version&quot;: 52}, &quot;44&quot;: {&quot;pk&quot;: 40512, &quot;version&quot;: 53}, &quot;45&quot;: {&quot;pk&quot;: 48949, &quot;version&quot;: 56}, &quot;46&quot;: {&quot;pk&quot;: 52131, &quot;version&quot;: 58}}"
```

Your approach is lean and mean and not prone to breakage, so I'm not sure whether this additional automation would be worth it.

Reply:
It would be indeed great If we could implement this cleanly.

Is there something like `yq` but for `html` files?

Maybe something like that can make it easier?

Reply:
> Is there something like yq but for html files?

`xq`

> Is xmllint what I basically ask?

It's an xml formatter and validator that apparently is both installed by default and can be used to parse HTML.

Reply:
I will resolve this for now, as it's too difficult to implement.